### ISSUE TITLE
Parse error with CASE statement

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#136]
Input:
```
SELECT
    DATE(zendesk.created_at, 'America/New_York') AS date,
    COUNT(
        CASE
        WHEN zendesk.support_team IN ('tech support', 'taskus', 'onc') THEN 1
        END
    ) AS tech_support
FROM
    zendesk
```

Results:
```
(sqlfluff-3.6.4) support_volume_forecast$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   6 | P:   9 |  PRS | Found unparsable segment @L006P009: 'END...'
L:   9 | P:   5 | L003 | Indentation not consistent with line #8
```

I'm pretty sure this is valid SQL -- it's from one of our projects and I "spot checked" it by pasting in the BigQuery web UI to confirm it didn't detect any syntax errors.

As a secondary, more minor issue, I'm not sure what the expected indentation is for the table name on line 9. When I split the `FROM` and table names across lines, I indent the table name this way.

**Comment by alanmcruickshank** [comment]
Good catch, silly error on my part there. PR going out now.
